presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/PartitionTable.java

```diff
@@ -164,10 +180,119 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         }
     }
 
+    public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint constraint)
+    {
+        requireNonNull(session, "session is null");
+        requireNonNull(tableHandle, "tableHandle is null");
+        requireNonNull(constraint, "constraint is null");
+        checkArgument(tableHandle instanceof IcebergTableHandle, "tableHandle not class IcebergTableHandle, but instead %s", tableHandle.getClass().getName());
```

Comment:
No need for this check. `ClassCastException` provides a good error message now.

Reply:
Removed.